### PR TITLE
Fixing destroy event

### PR DESCRIPTION
**Pull request by TigerC10**

### CODE REVIEW

**Comment by PowerKiKi** [comment]
v0.1.1 is now released and contains this fix

**Comment by odedniv** [comment]
So long time but I used this to solve my problem, which was incorrect.

Apparently the order of events hasn't changed, `$destroy` event is triggered on the element _before_ the DOM is removed [see $destroy documentation](https://docs.angularjs.org/api/ng/function/angular.element).

The problem is actually that this does not need to be implemented at all: jQuery UI intercepts element removals (probably just like angular does) and triggers the destroy of the element. Since it's interception is trigger before element `$destroy` is triggered we already have a destroyed slider.

You can see that by breakpointing the destroy function on the jQuery UI slider code. It works when using `scope.$on` because this is called before `element.remove`, and when jQuery UI's interception happen doesn't go through the method calling check the triggers the exception, but the destroy function is still called twice (try the breakpoint).

In conclusion, I think it's safe to remove these lines altogether (maybe in all angular-ui wrappers for jQuery UI? this looks like its not slider specific, but all widgets).

**Comment by mbordogna** [comment]
Just want to give the group a heads up, with the release of 1.5.9 of angular, the control is broke again.  Verified 1.5.8, and nothing is triggered after destruction.  However in 1.5.9 after the destroy is called, the following calls are made resulting in the before init errors 

line 172    elm.slider(method, ngModel.$viewValue);  --> triggered by line 177 ngModel.$render();
line 119    elm.slider('value', ui.values || ui.value);
line 71      elm.slider('option', newVal);

Seems like that 1.5.9 changed something related to the digest cycle.

Resulting errors 

angular.js:13947 Error: cannot call methods on slider prior to initialization; attempted to call method 'values'
    at Function.error (http://localhost:54323/app/build/vendor/jquery/dist/jquery.js:247:9)
    at HTMLDivElement.<anonymous> (http://localhost:54323/app/build/downloaded/jquery-ui/jquery-ui.custom.js:487:15)
    at Function.each (http://localhost:54323/app/build/vendor/jquery/dist/jquery.js:383:23)
    at jQuery.each (http://localhost:54323/app/build/vendor/jquery/dist/jquery.js:136:17)
    at jQuery.$.fn.(anonymous function) [as slider] (http://localhost:54323/app/build/downloaded/jquery-ui/jquery-ui.custom.js:483:9)
    at ngModel.$render (http://localhost:54323/app/build/vendor/angular-ui-slider/src/slider.js:172:25)
    at http://localhost:54323/app/build/vendor/angular-ui-slider/src/slider.js:177:33
    at Scope.$digest (http://localhost:54323/app/build/vendor/angular/angular.js:17654:23)
    at Scope.$apply (http://localhost:54323/app/build/vendor/angular/angular.js:17920:24)
    at $$debounceViewValueCommit (http://localhost:54323/app/build/vendor/angular/angular.js:28139:14)

-------------------------------------------

angular.js:13947 Error: cannot call methods on slider prior to initialization; attempted to call method 'option'
    at Function.error (http://localhost:54323/app/build/vendor/jquery/dist/jquery.js:247:9)
    at HTMLDivElement.<anonymous> (http://localhost:54323/app/build/downloaded/jquery-ui/jquery-ui.custom.js:487:15)
    at Function.each (http://localhost:54323/app/build/vendor/jquery/dist/jquery.js:383:23)
    at jQuery.each (http://localhost:54323/app/build/vendor/jquery/dist/jquery.js:136:17)
    at jQuery.$.fn.(anonymous function) [as slider] (http://localhost:54323/app/build/downloaded/jquery-ui/jquery-ui.custom.js:483:9)
    at http://localhost:54323/app/build/vendor/angular-ui-slider/src/slider.js:71:27
    at Scope.$digest (http://localhost:54323/app/build/vendor/angular/angular.js:17654:23)
    at Scope.$apply (http://localhost:54323/app/build/vendor/angular/angular.js:17920:24)
    at $$debounceViewValueCommit (http://localhost:54323/app/build/vendor/angular/angular.js:28139:14)
    at $setViewValue (http://localhost:54323/app/build/vendor/angular/angular.js:28111:12)
